cmd/sonobuoy/app/gen.go

```diff
@@ -77,3 +85,35 @@ func genManifest(cmd *cobra.Command, args []string) {
 	errlog.LogError(errors.Wrap(err, "error attempting to generate sonobuoy manifest"))
 	os.Exit(1)
 }
+
+// getRBACOrExit is a helper function for working with RBACMode. RBACMode is a bit of a special case
+// because it only needs a kubeconfig for detect, otherwise errors from kubeconfig can be ignored.
+// This function returns a bool because it os.Exit()s in error cases.
+func getRBACOrExit(mode *RBACMode, kubeconfig *Kubeconfig) bool {
+
+	// Usually we don't need a client. But in this case, we _might_ if we're using detect.
+	// So pass in nil if we get an error, then display the errors from trying to get a client
+	// if it turns out we needed it.
+	cfg, err := kubeconfig.Get()
+	var client *kubernetes.Clientset
+
+	var kubeError error
+	if err == nil {
+		client, err = kubernetes.NewForConfig(cfg)
+		if err != nil {
+			kubeError = err
+		}
+	} else {
+		kubeError = err
+	}
+
+	rbacEnabled, err := genFlags.rbacMode.Get(client)
+	if err != nil {
+		errlog.LogError(errors.Wrap(err, "couldn't detect RBAC mode."))
+		if errors.Cause(err) == RBACErrorNoClient {
+			errlog.LogError(kubeError)
+		}
+		os.Exit(1)
```

Comment:
Do we want to exit here? or make the utility more generic?

Reply:
The exit is because we're spitting out multiple errors, so it's not clear what we'd return.